### ISSUE TITLE
Navigator and large number of tile requests

### ISSUE THREAD

**Issue by thec0keman** [openseadragon/openseadragon#2610]
This really isn't a bug report, but we have some questions about how the navigator works with tile loading.

If a user is zoomed in on an image and clicks and drags in the navigator for several seconds, it is possible to generate a large number of tile requests.  (We've seen this easily jump into the hundreds).  This makes sense, as each ticket of the mouse drag the viewer is updated, and due to the zoom level + use of the navigator each tick will often move a significant enough distance to require fetching all or most tiles for the viewport.

This has the natural result that tile requests can back up, and depending on the tile response time can take a long time (> 10 seconds) to fully catch up to the user's current location. 

Is it possible with OSD for the navigator to debounce tile loading to prevent this?  I've been looking through the code and I'm not seeing an easy way to control this externally.

If not, is this something that would be worth adding as a feature?  For example, when creating a viewer, you could pass a parameter that controls a debounce start time for tile-loading while panning view the navigator.

**Comment by Aiosa** [comment]
Hi, there is no such a feature. It would have to be added, And yes, I believe this would be a worthy addition, but also a tricky one. You might end up with artifacts. Possible safe implementation would be to check for tiles in job queue, and if the tile is already outside viewport before it starts the job, abort such a job. Once a job is being issued, there is no way of stopping it.

**Comment by iangilman** [comment]
@thec0keman just checking, but you haven't modified `maxTilePerFrame` have you? Raising that number could potentially make this worse.

But yeah, this seems like a worthy thing to address somehow. It's not just dragging in the navigator that can cause a flood of tiles like this. Any sort of rapid panning can bring it on.

**Comment by thec0keman** [comment]
@iangilman Sorry for not following up on this.  We actually had modified `maxTilePerFrame` and reverting that significantly improved the behavior.   I'd be fine closing this for now if you'd prefer.

**Comment by iangilman** [comment]
@thec0keman Thanks for letting us know! Good to hear. 

I think we'll keep this issue open for now, just to consider the possibilities.

**Comment by bernard-crnkovic** [comment]
> Hi, there is no such a feature. It would have to be added, And yes, I believe this would be a worthy addition, but also a tricky one. You might end up with artifacts. Possible safe implementation would be to check for tiles in job queue, and if the tile is already outside viewport before it starts the job, abort such a job. Once a job is being issued, there is no way of stopping it.

Yes, it is tricky to stop issued requests, but what i believe OP meant was **avoiding** issuing request until viewport settles? My colleagues and I thought about something like debounce on `animation-start` and `animation-end` events however, we could not make it work because not enough of jobQueue API is exposed. in TIFY.

**Comment by Aiosa** [comment]
Yes tah could work, on the other side it sounds a bit hacky. The one driving tiles to load is the tiled image. A Simple option that loads tile from the destination viewport position rather than current one would be faster (tiles already preloaded when animation ends) and also cleaner to implement.
